src/libslic3r/GCode.cpp

```diff
@@ -1619,6 +1619,40 @@ static bool custom_gcode_sets_temperature(const std::string &gcode, const int mc
                     }
                 }
             }
+        } else if (*ptr == 'G' && include_g10) { // Only check for G10 if requested
```

Comment:
Shouldn't this test be done for the RepRap firmware only?

Reply:
Possibly, however that function does not have the print data available so that would need to be added to query the flavour. Do other flavours use G10 in a way that it would cause a problem? If it does not find a temperature just consumes the gcode and has no effect in any case. So unless another flavour uses G10 and the S parameter for something other than temperature I would not think the check is needed.

Reply:
the include_g10 parameter can be set false by the caller based on the firmware flavor.

Reply:
Yes, of course. I could make that change, will see what happens with the other discussion first.

Reply:
I believe the updated pull request resolves this issue.